devito/tools/dtypes_lowering.py

```diff
@@ -24,34 +24,70 @@
     "double": np.float64
 }
 
+
+def build_dtypes_vector(field_names, counts):
+    ret = {}
+    for base_name, base_dtype in mapper.items():
```

Comment:
`mapper` might be dangerously generic of a name with how much we use it internally

Reply:
I think it's fine here, it's used loosely for generic maps, and it's not exported to the outside modules of course, otherwise yes it'd be a pretty bad choice

Reply:
ok